internal/shardchain/dbfactory.go

```diff
@@ -34,3 +38,21 @@ type MemDBFactory struct{}
 func (f *MemDBFactory) NewChainDB(shardID uint32) (ethdb.Database, error) {
 	return rawdb.NewMemoryDatabase(), nil
 }
+
+// LDBShardFactory is a merged Multi-LDB-backed blockchain database factory.
+type LDBShardFactory struct {
+	RootDir    string // directory in which to put shard databases in.
+	DiskCount  int
+	ShardCount int
+}
+
+// NewChainDB returns a new memDB for the blockchain for given shard.
+func (f *LDBShardFactory) NewChainDB(shardID uint32) (ethdb.Database, error) {
+	dir := filepath.Join(f.RootDir, fmt.Sprintf("harmony_sharddb_%d", shardID))
```

Comment:
make the sharddb prefix configurable

Reply:
My suggestion is to use a fixed value, because the two data formats are incompatible with the other, it can also prevent users from making changes by mistake, and it is more convenient to write subsequent documents

Reply:
okay